purchases/src/main/kotlin/com/revenuecat/purchases/models/GoogleStoreProduct.kt

```diff
@@ -1,10 +1,21 @@
 package com.revenuecat.purchases.models
 
 import com.android.billingclient.api.ProductDetails
+import com.revenuecat.purchases.PresentedOfferingContext
 import com.revenuecat.purchases.ProductType
 import java.util.Locale
 
-data class GoogleStoreProduct(
+data class GoogleStoreProduct
+@JvmOverloads
+@Deprecated(
```

Comment:
Quick note about this deprecation. I was running into a lot of issues trying to add the new parameter while trying to keep retrocompatibility... In the end, I'm deprecating the main constructor and adding a new one which is internal only. 

Reason was that there were a lot of overload conflicts, specially in Java, where String can be null and it was making the API tests fail. 

IMO, we should make all these constructors internal in the future. If we want to allow devs to create them, then I would suggest using the `Builder` pattern instead... Much easier to add/deprecate things and allows us to keep flexibility for the constructor.